autovalet_gazebo/scripts/validate_parking.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+# license removed for brevity
+
+# Author: Eva Mo
+# Adapted from: validate_arucotags.py
+# Date of creation: October 9, 2020 04:20 PM
+# Description: Validation script for Progress Review 9, compare husky final pose with arucotags gt.pose + 4m into the right spot
+
+import os
+import numpy as np
+
+import rospy
+from gazebo_msgs.msg import LinkStates
+from tf.transformations import euler_from_quaternion
+
+class parking_pose_validation:
+    def __init__(self, gt_topic):
+        # gt: ground truth published by gazebo link state
+        self.gt = rospy.wait_for_message(gt_topic, LinkStates)
+        self.aruco_tag_gt = None
+        self.base_link_gt = None
+
+        self.target_pose = self.get_target_pose()
+        self.reached_pose = self.get_reached_pose()
+        self.calculate_error()
+
+    def get_target_pose(self):
+        self.aruco_tag_gt    = self.gt.pose[-7] # name 'aruco_visual_marker_7::marker'
+        # x += 4 compensating for the translation between parking spot & tag
+        aruco_tag_gt_2d = np.array([self.aruco_tag_gt.position.x , self.aruco_tag_gt.position.y - 4, self.quat_to_euler(self.aruco_tag_gt.orientation)[2]])
+
+        return aruco_tag_gt_2d
+
+    def get_reached_pose(self):
+        self.base_link_gt    = self.gt.pose[-5] # base_link w.r.t the gazebo world origin
+        base_link_gt_2d = np.array([self.base_link_gt.position.x, self.base_link_gt.position.y, self.quat_to_euler(self.base_link_gt.orientation)[2]])
+
+        return base_link_gt_2d
+
+    def calculate_error(self):
+        # Error in x and y in cms
+        trans_err = (self.target_pose[:-1] - self.reached_pose[:-1])*100
+        # Error in orientation (yaw) in degrees 
+        yaw_err   = np.abs((self.target_pose[2]*180/3.14 - 270) - self.reached_pose[2]*180/3.14) #comparing the aruco tag (rotated by 270deg) with husky's yaw
```

Comment:
will it always be 270? make a note maybe that we have to have the tag at a specific orientation for it to work. we'd have to mess around with the real arucotag on HW to make sure the orientation is right for this code to work

Reply:
comment added